File: backend/resources/export.go
package resources

import "time"
import "github.com/stashsphere/backend/models"

type ExportStatus struct {
	ID        string     `json:"id"`
	Status    string     `json:"status"`
	CreatedAt time.Time  `json:"createdAt"`
	ExpiresAt *time.Time `json:"expiresAt"`
	Error     *string    `json:"error"`
}

func ExportStatusFromModel(e *models.Export) ExportStatus {
	return ExportStatus{
		ID:        e.ID,
		Status:    string(e.Status),
		CreatedAt: e.CreatedAt,
		ExpiresAt: e.ExpiresAt.Ptr(),
		Error:     e.ErrorMSG.Ptr(),
	}
}

File: frontend/src/api/export.ts
import { Axios } from 'axios';

export type ExportStatus = {
  id: string;
  status: string;
  createdAt: string;
  expiresAt: string | null;
  error: string | null;
};

export const triggerExport = async (axios: Axios): Promise<ExportStatus> => {
  const response = await axios.post(
    '/export',
    {},
    {
      headers: { 'Content-Type': 'application/json' },
    }
  );
  return response.data as ExportStatus;
};

export const getExportStatus = async (axios: Axios): Promise<ExportStatus | null> => {
  const response = await axios.get('/export', {
    headers: { 'Content-Type': 'application/json' },
    validateStatus: (status) => status === 200 || status === 404,
  });
  if (response.status === 404) return null;
  return response.data as ExportStatus;
};
